### PR TITLE
Complete rework of WhisperHandler including proper chase functionality

### PR DESCRIPTION
**Pull request by RubenD96**
This PR focuses on WhisperHandler.
Some of the heavenms code was re-used (anti-spam, logging), the rest was all rewritten to be more readable.
Also includes proper chase functionality, admins can use "/c name" to teleport to the player's exact location.
WHISPER packets were redone to have proper flags.

### CODE REVIEW

File: src/main/java/net/server/channel/handlers/WhisperHandler.java [hunk 1]
```diff
@@ -26,121 +26,85 @@
...
-                            c.announce(MaplePacketCreator.getWhisperReply(recipient, (byte) 1));
-                    } else {
-                        c.announce(MaplePacketCreator.getWhisperReply(recipient, (byte) 0));
-                    }
+    public void handlePacket(SeekableLittleEndianAccessor slea, MapleClient c) {
+        byte request = slea.readByte();
+        String name = slea.readMapleAsciiString();
+        MapleCharacter target = c.getWorldServer().getPlayerStorage().getCharacterByName(name);
+
+        if (target != null) {
```
Comment (Arnuh) [inline src/main/java/net/server/channel/handlers/WhisperHandler.java:52]: It's cleaner to just do a null check, give error, and return early. It's used in other places for anti cheat checks so it should be fine for standards.

File: src/main/java/tools/MaplePacketCreator.java [hunk 2]
```diff
@@ -6349,43 +6330,64 @@ private static void addPetInfo(final MaplePacketLittleEndianWriter mplew, MapleP
                 return showCash(mc);
         }
 
+        public static class WhisperFlag {
+                public static final byte LOCATION = 0x01;
+                public static final byte WHISPER = 0x02;
+                public static final byte REQUEST = 0x04;
+                public static final byte RESULT = 0x08;
+                public static final byte RECIEVE = 0x10;
```
Comment (P0nk) [inline src/main/java/tools/MaplePacketCreator.java:6338]: Misspelled "RECEIVE"

File: src/main/java/tools/MaplePacketCreator.java [hunk 3]
```diff
@@ -3661,31 +3667,6 @@ private static void writeLongMaskChair(final MaplePacketLittleEndianWriter mplew
                 return pOutPacket.getPacket();
         }
 
-        public static byte[] getWhisper(String sender, int channel, String text) {
```
Comment (P0nk) [inline src/main/java/tools/MaplePacketCreator.java:3664]: Can't compile due to this renaming to `getWhisperReceive`. Issue is in method `net.server.world.World#whisper` which uses the old method.

File: src/main/java/tools/MaplePacketCreator.java [hunk 4]
```diff
@@ -6349,43 +6330,64 @@ private static void addPetInfo(final MaplePacketLittleEndianWriter mplew, MapleP
...
+        }
+
+        public static byte[] getWhisperReceive(String sender, int channel, boolean fromAdmin, String message) {
+                MaplePacketLittleEndianWriter mplew = new MaplePacketLittleEndianWriter();
+                mplew.writeShort(SendOpcode.WHISPER.getValue());
+
+                mplew.write(WhisperFlag.WHISPER | WhisperFlag.RECIEVE);
+                mplew.writeMapleAsciiString(sender);
+                mplew.write(channel);
+                mplew.writeBool(fromAdmin);
```
Comment (P0nk) [inline src/main/java/tools/MaplePacketCreator.java:6388]: Strange that it worked before without this `writeBool`... (just a comment)

**Review by P0nk — CHANGES_REQUESTED** [review]
A lot of noise in the PR due to formatting changes, which made it more difficult to review. 
Did some successful manual tests after fixing the compile issue locally, so it looks good. You just need to fix some minor issues.

**Review by P0nk — APPROVED** [review]

**Pull request merged**